acadopy/bindings.pyx

```diff
@@ -717,7 +718,9 @@ cdef class MultiObjectiveAlgorithm(OptimizationAlgorithm):
         print('called mco init')
 
     def __dealloc__(self):
-        del self._thisptr
+        if self._owner:
```

Comment:
I would suggest testing if the pointer is null in place of the ownership. Otherwise, the logic gets broken.

Reply:
Good idea, I'll do the same in the __dealloc__ for every class.